### PR TITLE
Sid extension

### PR DESCRIPTION
**Pull request by ltn22**
add --sid-extension to add to the sid file some extra element to allows a better manipulation on constrained devices. See README.md

### CODE REVIEW

**Comment by ltn22** [comment]
this is done in branch sid-extension

File: pyang/plugins/sid.py [hunk 1]
```diff
@@ -221,6 +228,11 @@ def print_help():
 
   $ pyang --sid-check-file toaster@2009-12-28.sid toaster@2009-12-28.yang
 
+--sid-extenstion
```
Comment (cabo) [inline pyang/plugins/sid.py:231]:
```suggestion
--sid-extension
```

File: pyang/plugins/sid.py [hunk 2]
```diff
@@ -575,6 +588,9 @@ def collect_module_items(self, module):
         if 'item' not in self.content:
             self.content['item'] = []
 
+        if 'key-mapping' not in self.content:
```
Comment (cabo) [inline pyang/plugins/sid.py:591]: Does that leave an empty entry if --sid-extension is not given?

Comment (cabo) [reply pyang/plugins/sid.py:598]: sorry, this is not too easy for me to set up a test for

Comment (ltn22) [reply pyang/plugins/sid.py]: You're right, the key-mapping was appearing even if sid-extension was not set.

I fixed it line 598,

**Pull request merged**